src/azure-cli-core/azure/cli/core/tests/test_profile.py

```diff
@@ -1261,35 +1251,45 @@ def test_get_raw_token_in_cloud_console(self, mock_msi_auth, mock_in_cloud_conso
         consolidated[0]['user']['cloudShellID'] = True
         profile._set_subscriptions(consolidated)
 
-        mi_auth_instance = None
+        # The below code creates a credential instance and checks it.
+        #
+        # We can define a normal variable `credential_instance` here and use `nonlocal` to assign the credential
+        # instance to it, but using a mutable list also allows us to check how many instances are created.
+        # See https://stackoverflow.com/a/8448011/2199657
+        #
+        # test_login_in_cloud_shell retrieves the credential instance from
+        # create_subscription_client_mock.call_args.args[1], so another possible way to retrieve the credential
+        # instance is to create a hook in get_raw_token and patch that hook during tests.
+        credential_instances = []
 
-        def mi_auth_factory(*args, **kwargs):
-            nonlocal mi_auth_instance
-            mi_auth_instance = MSRestAzureAuthStub(*args, **kwargs)
-            return mi_auth_instance
+        def cloud_shell_credential_factory():
+            credential = CloudShellCredentialStub()
+            credential_instances.append(credential)
+            return credential
 
-        mock_msi_auth.side_effect = mi_auth_factory
+        cloud_shell_credential_mock.side_effect = cloud_shell_credential_factory
 
         # action
-        cred, subscription_id, tenant_id = profile.get_raw_token(resource=self.adal_resource)
+        token_tuple, subscription_id, tenant_id = profile.get_raw_token(scopes=self.msal_scopes)
```

Comment:
The name `cred` conflicts with the `CloudShellCredential` concept, so I rename it to better reflect its content.